py_gtfs_rt_ingestion/ingest.py

```diff
@@ -9,11 +9,121 @@
 import pyarrow.parquet as pq
 
 from datetime import datetime
+from multiprocessing import Pool
+from multiprocessing import Queue
 from pathlib import Path
 
 from py_gtfs_rt_ingestion import Configuration
 
+
 DESCRIPTION = "Convert a json file into a parquet file. Used for testing."
+MULTIPROCESSING_POOL_SIZE = 4
+
+def gz_to_pyarrow(filename: str, config: Configuration, return_queue: Queue) -> None:
+    """
+    Accepts filename as string. Converts gzipped json -> pyarrow table. 
+
+    parrow table sent back to main process using return_queue.
+    If conversion fails, returns filename.
+    """
+    # Enclose entire function in try block, Process can't fail silently.
+    # Must return either pyarrow table or filename
+    try:
+        with gzip.open(filename, 'rb') as f:
+            json_data = json.loads(f.read())
+
+        # parse timestamp info out of the header
+        header = json_data['header']
+        feed_timestamp = header['timestamp']
+        timestamp = datetime.utcfromtimestamp(feed_timestamp)
+
+        # create dict of lists 
+        ret_obj = {key.name:[] for key in config.export_schema}
+
+        # for each entity in the list, create a record and add it to the table
+        for entity in json_data['entity']:
+            record = config.record_from_entity(entity=entity)
+            record.update({
+                'year': timestamp.year,
+                'month': timestamp.month,
+                'day': timestamp.day,
+                'hour': timestamp.hour,
+                'feed_timestamp': feed_timestamp
+            })
+
+            for key in record:
+                ret_obj[key].append(record[key])
+
+        ret_obj = pa.table(ret_obj, schema=config.export_schema)
+
+    # Catch failure, send back filename for handling
+    except Exception as e:
+        print(e, flush=True)
+        ret_obj = filename
+
+    # Send pyarrow table or filename back to main Process for concatenation
+    return_queue.put(ret_obj)
+    return_queue.close()
+
+def lambda_handler(event: dict, context) -> None:
```

Comment:
it would be kind of cool if this is what we were calling when we invoke the script too.